services/app-web/src/pages/StateSubmission/ReviewSubmit/ReviewSubmitModal.tsx

```diff
@@ -0,0 +1,177 @@
+                'You must provide a summary of changes'
+            ),
+        }),
+        onSubmit: (values) => onModalSubmit(values),
+    })
+
+    const toggleIsSubmitting = (
+        previousState: boolean,
+        currentState: boolean
+    ) => {
+        if (currentState !== previousState) {
+            isSubmitting(currentState)
+            return currentState
+        }
+        return previousState
+    }
+
+    const [submittingState, dispatch] = useReducer(toggleIsSubmitting, false)
+
+    const submitHandler = async () => {
+        setFocusErrorsInModal(true)
+        if (unlocked) {
+            formik.handleSubmit()
+        } else {
+            await onSubmit(undefined)
+        }
+    }
+
+    const onModalSubmit = async (values: typeof modalFormInitialValues) => {
+        const { submittedReason } = values
+        await onSubmit(submittedReason)
+    }
+
+    const onSubmit = async (
+        submittedReason: string | undefined
+    ): Promise<void> => {
+        const input = { pkgID: draftSubmission.id }
+
+        if (unlocked) {
+            Object.assign(input, {
+                submittedReason: submittedReason,
+            })
+        }
+
+        try {
+            const data = await submitDraftSubmission({
+                variables: {
+                    input: input,
+                },
+            })
+
+            if (data.errors) {
+                showError('Error attempting to submit. Please try again.')
+                modalRef.current?.toggleModal(undefined, false)
+            }
+
+            if (data.data?.submitHealthPlanPackage) {
+                modalRef.current?.toggleModal(undefined, false)
+                navigate(`/dashboard?justSubmitted=${submissionName}`)
+            } else {
+                console.error('Got nothing back from submit')
+                showError('Error attempting to submit. Please try again.')
+                modalRef.current?.toggleModal(undefined, false)
+            }
+        } catch (error) {
+            showError('Error attempting to submit. Please try again.')
+            modalRef.current?.toggleModal(undefined, false)
+        }
+    }
+
+    // Focus submittedReason field in submission modal on Resubmit click when errors exist
+    useEffect(() => {
+        if (focusErrorsInModal && formik.errors.submittedReason) {
+            const fieldElement: HTMLElement | null = document.querySelector(
+                `[name="submittedReason"]`
+            )
+
+            if (fieldElement) {
+                fieldElement.focus()
+                setFocusErrorsInModal(false)
+            } else {
+                console.log('Attempting to focus element that does not exist')
+            }
+        }
+    }, [focusErrorsInModal, formik.errors])
+
+    useEffect(() => {
+        dispatch(formik.isSubmitting || submitMutationLoading)
+    }, [formik.isSubmitting, submitMutationLoading])
+
+    return (
+        <Modal
```

Comment:
I'm not sure the reasoning for using `useFormik` hook instead of the component, I just tried to follow the same pattern as `UnlockModal` component. @haworku Any particular reason? simplicity?

Reply:
Ha! i’m starting to learn that as I’m working on the FormYesNo component. I need to read more docs but this now makes sense!